groovy-language-server/src/main/java/com/palantir/ls/groovy/compilation/GroovyTreeParser.java

```diff
@@ -66,163 +53,107 @@
-import org.codehaus.groovy.control.messages.Message;
-import org.codehaus.groovy.control.messages.SyntaxErrorMessage;
-import org.codehaus.groovy.control.messages.WarningMessage;
-import org.codehaus.groovy.syntax.SyntaxException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-/**
- * Wraps the Groovy compiler and provides Language Server Protocol diagnostics on compile.
- */
-public final class GroovycWrapper implements CompilerWrapper {
+public final class GroovyTreeParser implements TreeParser {
 
-    private static final Logger logger = LoggerFactory.getLogger(GroovycWrapper.class);
+    private static final Logger logger = LoggerFactory.getLogger(GroovyTreeParser.class);
 
     private static final String GROOVY_DEFAULT_INTERFACE = "groovy.lang.GroovyObject";
     private static final String JAVA_DEFAULT_OBJECT = "java.lang.Object";
 
-    private final Path workspaceRoot;
-    private final Path changedFilesRoot;
-    private final CompilerConfiguration config;
-
-    private CompilationUnit unit;
-
     // Maps from source file path -> set of symbols in that file
     private Map<URI, Set<SymbolInformation>> fileSymbols = Maps.newHashMap();
     // Maps from type -> set of references of this type
     private Map<String, Set<SymbolInformation>> typeReferences = Maps.newHashMap();
-    // Map from origin source filename to its changed version source writer
-    private Map<URI, SourceWriter> originalSourceToChangedSource = Maps.newHashMap();
 
-    private GroovycWrapper(Path workspaceRoot, Path changedFilesRoot, CompilerConfiguration config) {
+    private final CompilationUnitProvider unitProvider;
+    private final Path workspaceRoot;
+    private final Path changedFilesRoot;
+
+
+    private GroovyTreeParser(CompilationUnitProvider unitProvider, Path workspaceRoot, Path changedFilesRoot) {
+        this.unitProvider = unitProvider;
         this.workspaceRoot = workspaceRoot;
         this.changedFilesRoot = changedFilesRoot;
-        this.config = config;
-        unit = new CompilationUnit(config);
     }
 
     /**
-     * Creates a new instance of GroovycWrapper.
+     * Creates a new instance of GroovyTreeParser.
      *
-     * @param targetDirectory the directory in which to put generated files
+     * @param unitProvider the compilation unit provider in which to store our current unit
      * @param workspaceRoot the directory to compile
      * @param changedFilesRoot the directory in which to temporarily store incrementally changed files
-     * @return the newly created GroovycWrapper
+     * @return the newly created GroovyTreeParser
      */
-    public static GroovycWrapper of(Path targetDirectory, Path workspaceRoot, Path changedFilesRoot) {
-        checkNotNull(targetDirectory, "targetDirectory must not be null");
+    public static GroovyTreeParser of(CompilationUnitProvider unitProvider, Path workspaceRoot, Path changedFilesRoot) {
```

Comment:
`GroovyTreeParser` shouldn't need `changedFilesRoot`. I need to figure out another way to resolve uri in `getWorkspaceUri`